source/common/quic/envoy_quic_client_session.cc

```diff
@@ -13,15 +15,40 @@
 namespace Envoy {
 namespace Quic {
 
+// Quic connection doesn't use transport socket object to do handshake. This callback
+// implementation is used by the cert verifier to access Network::Connection interfaces.
```

Comment:
How about something like:

// This class is only used to provide extra context during certificate validation. As such
// it does not implement the various methods which are irrelevant to certificate validation.